### PR TITLE
remove singletons

### PR DESCRIPTION
**Pull request by bettybao1209**
~~need to add a new msg type `GetInstance `  in `LocalNode` class like this:~~
```
protected override void OnReceive(object message)
        {
            base.OnReceive(message);
            switch (message)
            {
                case Message msg:
                    BroadcastMessage(msg);
                    break;
                case RelayDirectly relay:
                    OnRelayDirectly(relay.Inventory);
                    break;
                case SendDirectly send:
                    OnSendDirectly(send.Inventory);
                    break;
                case GetInstance _:
                    Sender.Tell(this);
                    break;
            }
        }
```

### CODE REVIEW

File: neo-cli/CLI/MainService.Node.cs [hunk 1]
```diff
@@ -69,14 +69,15 @@ private void OnShowStateCommand()
...
                 while (!cancel.Token.IsCancellationRequested)
                 {
-                    uint height = NativeContract.Ledger.CurrentIndex(Blockchain.Singleton.View);
-                    uint headerHeight = Blockchain.Singleton.HeaderCache.Last?.Index ?? height;
+                    uint height = NativeContract.Ledger.CurrentIndex(NeoSystem.StoreView);
+                    uint headerHeight = NeoSystem.HeaderCache.Last?.Index ?? height;
 
                     Console.SetCursorPosition(0, 0);
-                    WriteLineWithoutFlicker($"block: {height}/{headerHeight}  connected: {LocalNode.Singleton.ConnectedCount}  unconnected: {LocalNode.Singleton.UnconnectedCount}", Console.WindowWidth - 1);
+                    LocalNode localNode = await NeoSystem.LocalNode.Ask<LocalNode>(new LocalNode.GetInstance());
```
Comment (ProDog) [inline neo-cli/CLI/MainService.Node.cs:76]: LocalNode.GetInstance() compile fail.

Comment (bettybao1209) [reply neo-cli/CLI/MainService.Node.cs:76]: It needs to add the message type in neo-core, which is to be determined.

**Comment by erikzhang** [comment]
Please check neo-project/neo#2311

**Review by cloud8little — APPROVED** [review]

**Comment by cloud8little** [comment]
test passed.

**Review by shargon — APPROVED** [review]

**Pull request merged**